### DISCUSSION TITLE
pytorch_lightning.utilities.exceptions.MisconfigurationException: GPUAccelerator can not run on your system since the accelerator is not available. The following accelerator(s) is available and can be passed into `accelerator` argument of `Trainer`: ['cpu'].

### DISCUSSION THREAD

**Discussion by uiop2020** [Lightning-AI/pytorch-lightning#13255] [code help: CV]
I am receiving this error, but I have GPU on my device. what should I do to be able to use GPU?

pytorch_lightning.utilities.exceptions.MisconfigurationException: GPUAccelerator can not run on your system since the accelerator is not available. The following accelerator(s) is available and can be passed into `accelerator` argument of `Trainer`: ['cpu'].

**Comment by akihironitta** [comment]
Hi @uiop2020 PyTorch Lightning simply uses `torch.cuda.device_count()` to detect if any GPU(s) is available:
https://github.com/PyTorchLightning/pytorch-lightning/blob/bfa8b7be2d99b980afa62f5cb0433326bcfd2ef0/pytorch_lightning/accelerators/gpu.py#L90-L91

so, I believe that's an issue with your environment. To address your issue, you might want to read the threads linked in [this comment](https://discuss.pytorch.org/t/torch-c-cuda-getdevicecount-0-returns-false/113864/2?u=akihironitta).

**Comment by rehammsalah** [comment]
did you solve this problem?
cuz i have the same error

**Reply by akihironitta** [reply]
What is the output of `torch.cuda.device_count()` in your env?

**Reply by kaan-amani** [reply]
This is how it looks like normally
<img width="279" alt="image" src="https://github.com/Lightning-AI/pytorch-lightning/assets/85875540/6ab5d5fc-5ae3-4632-96f9-8627877fa068">

But after I run my training script, which was working fine until last week, gives this error and changes the output for device count
<img width="1442" alt="image" src="https://github.com/Lightning-AI/pytorch-lightning/assets/85875540/ff8295d4-6523-48c9-8df0-c4042968547b">

**Comment by Huiflorazhan** [comment]
> What is the output of `torch.cuda.device_count()` in your env?

my output is 0. Could you help me with this problem?

**Reply by jxtngx** [reply]
can you please run the following command in terminal, and post a screenshot of the result here.

```sh
nvidia-smi
```

**Reply by Huiflorazhan** [reply]
The output is command not found. I'm using Mac.

**Reply by jxtngx** [reply]
okay, that explains it.  when on mac, we need to set `accelerator` to `mps` instead of `gpu`. and `devices` should be 1.

**Reply by Huiflorazhan** [reply]
Could you please help me with this issue? Thanks~
https://github.com/scverse/scvi-tools/issues/2246
